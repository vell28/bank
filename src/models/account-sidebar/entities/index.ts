export interface IAccountSidebar {
  [key: string]: boolean;
}
