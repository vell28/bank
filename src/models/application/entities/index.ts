export interface IApplication {
  applicationId: string;
  productId: string;
  Authorization: string;
  deviceId?: string;

  readonly bankName: string;
  readonly BIC: string;
}
