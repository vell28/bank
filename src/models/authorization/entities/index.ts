export interface IAuthorization {
  service_code?: string;
  phone?: string;
  token?: string;
  sessid?: string;
  token_expires?: Date;
  // maybe its not required
  'dev-id'?: string;
  'token-firebase'?: string;
  isShownLogoutConfirmModal: boolean;
}

export interface ITokenRefresh {
  success: boolean;
  token: string;
}

export interface ISuccessLogout {
  success: boolean;
}

export interface ICodeRequest {
  sessid: string;
  code: string | null;
  success?: boolean;
}

export interface IPasswordCheckRequest {
  password: string;
  phone: string;
}

export interface ISuccessCheckResponse {
  status: 'ok';
}
