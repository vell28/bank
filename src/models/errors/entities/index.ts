export type AppErrorType = '401' | '404' | '403' | '500' | 'ERR_INTERNET_DISCONNECTED';

export interface IAppErrors {
  title?: string;
  subtitle?: string;
  onOk?: () => void;
  btnText?: string;
  supportUrl?: string;
  type?: AppErrorType;
  meta?: string;
}
