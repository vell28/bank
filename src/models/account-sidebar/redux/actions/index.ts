import { getStateType, IAction } from 'modules/store/types';
import { Dispatch } from 'redux';
import { push } from 'connected-react-router';

import { BANKING_PATH } from 'containers/routing/utils';
import { getRouterPathname } from 'models/router/selectors';
import { IAccountSidebar } from '../../entities';

export const UPDATE_ACCOUNT_SIDEBAR = 'sidebar/UPDATE_ACCOUNT';
export const SHOW_HIDE_SIDEBAR = 'sidebar/TOGGLE';

export type IUpdateSidebarAction = IAction<IAccountSidebar>;
export type IToggleSidebar = IAction<boolean>;

const BANKING_ACCOUNT = '/banking/account/';

export const updateSidebar = (data: IAccountSidebar): IUpdateSidebarAction => ({
  type: UPDATE_ACCOUNT_SIDEBAR,
  payload: data,
});

export const getSidebarAccountId = (data: IAccountSidebar) => (dispatch: Dispatch, setState: getStateType): any => {
  const pathname = getRouterPathname(setState());

  const path = pathname.replace(BANKING_ACCOUNT, '').replace('/', '');

  if (path && !data[path] && data[path] !== undefined) {
    dispatch(push(BANKING_PATH));
  }

  dispatch(updateSidebar(data));
};

export const toggleSidebar = (isShown: boolean): IToggleSidebar => ({
  type: SHOW_HIDE_SIDEBAR,
  payload: isShown,
});
