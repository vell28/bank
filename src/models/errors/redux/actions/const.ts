import { defaultTo } from 'ramda';
import { push } from 'connected-react-router';

import { store } from 'modules/store';
import { PATHS_CONFIG } from 'containers/routing/utils';

import { IAppErrors } from '../../entities';

export const default404 = defaultTo<IAppErrors>({
  type: '404',
  title: 'Page not found',
  subtitle: 'error: 404',
});

export const default500 = defaultTo<IAppErrors>({
  type: '500',
  title: 'Sorry, something went wrong.\n Please try again or contact support.',
  subtitle: 'error: 500',
  supportUrl: '/',
  btnText: 'Try again',
});

export const default403 = defaultTo<IAppErrors>({
  type: '403',
  title: 'We are sorry, but you do not have access to this page',
  subtitle: 'error: 403',
});

export const defaultInternet = defaultTo<IAppErrors>({
  type: 'ERR_INTERNET_DISCONNECTED',
  title: 'Network Error. Please check your network connection and try again.',
  subtitle: 'error: No Internet.',
  btnText: 'Try again',
});

export const expiredSession: IAppErrors = {
  type: '401',
  title: 'Your session has expired.',
  btnText: 'Ok',
  onOk: () => store.dispatch(push(PATHS_CONFIG.authorization.path)),
};
