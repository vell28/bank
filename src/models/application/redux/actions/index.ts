import { fetchOrganizations } from 'models/organizations/redux/actions';
import { resetTransferLoadingState } from 'models/operations/transfer/redux/actions';
import { fetchTransactionsPage } from 'models/transactions/redux/actions';
import { isNotEmpty } from 'utils/ramda';

import { getOrganizations, getCurrentAccount, getClient } from 'models/organizations/redux/selectors';
import { getStateType } from 'modules/store/types';
import { ThunkAction } from 'utils/utilTypings';
import { saveAccountDetails, copyToAccountDetails } from 'modules/pdf/templates/accountDetails';
import { propOr } from 'ramda';
import { getBankName, getBankBIС } from '../..';

export const INIT_APP = 'application/SET';
export const UPDATE_APP = 'application/UPDATE';

export const initApp = async (dispatch: any, getState: getStateType) => {
  try {
    const organizations = getOrganizations(getState());
    // if organizations exist, fetch history and update organizations in backgound
    if (isNotEmpty(organizations)) {
      dispatch(fetchOrganizations());
      dispatch(fetchTransactionsPage());
    } else {
      await dispatch(fetchOrganizations());
      dispatch(fetchTransactionsPage());
    }
  } catch (e) {
    throw e;
  }
  dispatch(resetTransferLoadingState);
};

export const exportAccountDetails = (): ThunkAction => {
  return (dispatch: any, getState: getStateType) => {
    const state = getState();
    const account = getCurrentAccount(state);
    const bankTitle = getBankName(state);
    const bankTin = getBankBIС(state);
    const client = getClient(state);

    const iban: string = propOr('', 'id')(account);
    const recipientName: string = propOr('', 'name')(client);

    saveAccountDetails(iban, recipientName, bankTitle, bankTin);
  };
};

export const copyAccountDetails = (): ThunkAction => {
  return (dispatch: any, getState: getStateType) => {
    const state = getState();
    const account = getCurrentAccount(state);
    const bankTitle = getBankName(state);
    const bankTin = getBankBIС(state);
    const client = getClient(state);

    const iban: string = propOr('', 'id')(account);
    const recipientName: string = propOr('', 'name')(client);

    copyToAccountDetails(iban, recipientName, bankTitle, bankTin);
  };
};
