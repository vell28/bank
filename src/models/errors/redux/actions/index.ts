import { Dispatch } from 'redux';
import { push } from 'connected-react-router';

import { ERROR_PATH } from 'containers/routing/utils';
import { IAction } from 'modules/store/types';
import {
  default404, default500, default403, defaultInternet, expiredSession
} from './const';

import { IAppErrors } from '../../entities';

export const SET_APP_ERROR = 'app-error/SET';

export type IAppErrorAction = IAction<IAppErrors>;

export const app404Error = (dispatch: Dispatch) => (error?: IAppErrors) => {
  dispatch({
    type: SET_APP_ERROR,
    payload: default404(error),
  });
  dispatch(push(ERROR_PATH));
};

export const app500Error = (dispatch: Dispatch) => (error?: IAppErrors) => {
  dispatch({
    type: SET_APP_ERROR,
    payload: default500(error),
  });
  dispatch(push(ERROR_PATH));
};

export const app403Error = (dispatch: Dispatch) => (error?: IAppErrors) => {
  dispatch({
    type: SET_APP_ERROR,
    payload: default403(error),
  });
  dispatch(push(ERROR_PATH));
};

export const appInternetError = (dispatch: Dispatch) => (error?: IAppErrors) => {
  dispatch({
    type: SET_APP_ERROR,
    payload: defaultInternet(error),
  });
  dispatch(push(ERROR_PATH));
};

export const appSessionExpired = (dispatch: Dispatch) => {
  dispatch({
    type: SET_APP_ERROR,
    payload: expiredSession,
  });
  dispatch(push(ERROR_PATH));
};
