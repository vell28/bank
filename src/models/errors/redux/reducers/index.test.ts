import { appErrors, IAppErrorsState } from '.';
import { SET_APP_ERROR, IAppErrorAction } from '../actions';
import { IAppErrors } from '../../entities';

const initMainMenu: IAppErrorsState = {};

const data: IAppErrors = {
  type: '404',
  title: 'Page not found',
  subtitle: 'error: 404',
};

describe('appErrors reducers', () => {
  test('mainModal/UPDATE_MAIN_MODAL', () => {
    const action: IAppErrorAction = {
      type: SET_APP_ERROR,
      payload: data,
    };

    const newState = appErrors(initMainMenu, action);
    expect(newState).toMatchSnapshot();
  });
});
