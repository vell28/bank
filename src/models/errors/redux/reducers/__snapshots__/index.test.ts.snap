// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appErrors reducers mainModal/UPDATE_MAIN_MODAL 1`] = `
Object {
  "subtitle": "error: 404",
  "title": "Page not found",
  "type": "404",
}
`;
